src/micro_http/src/server.rs

```diff
@@ -379,6 +380,22 @@ impl HttpServer {
         Ok(parsed_requests)
     }
 
+    /// This function is responsible with flushing any remaining outgoing
+    /// requests on the server.
+    ///
+    /// Note that this function can block the thread on write, since the
+    /// operation is blocking.
+    pub fn flush_outgoing_writes(&mut self) {
+        for (_, connection) in self.connections.iter_mut() {
+            while connection.state == ClientConnectionState::AwaitingOutgoing {
+                if let Err(e) = connection.write() {
+                    error!("Connection write error: {}", e);
```

Comment:
I suggest avoiding the logging with `error!` if the error we receive from `connection.write()` is `Err(ServerError::ConnectionError(ConnectionError::InvalidWrite))`, since that would indicate that we have nothing to write, so we have successfully flushed everything.

Reply:
I agree with @georgepisaltu . Any other error is caught inside the write function. If we still want to log an error, the actual logging could be moved inside the write function itself. What do you think?

Reply:
Done!